instrumentation/action_mailer/README.md

```diff
@@ -0,0 +1,102 @@
+# OpenTelemetry ActionMailer Instrumentation
+
+The ActionMailer instrumentation is a community-maintained instrumentation for the ActionMailer portion of the [Ruby on Rails][rails-home] web-application framework.
+
+## How do I get started?
+
+Install the gem using:
+
+```bash
+# Install just the ActionMailer instrumentation
+gem install opentelemetry-instrumentation-action_mailer
+# Install the ActionMailer instrumentation along with the rest of the Rails-related instrumentation
+gem install opentelemetry-instrumentation-rails
+```
+
+Or, if you use [bundler][bundler-home], include `opentelemetry-instrumentation-action_mailer` in your `Gemfile`.
+
+## Usage
+
+To use the instrumentation, call `use` with the name of the instrumentation:
+
+```ruby
+OpenTelemetry::SDK.configure do |c|
+  # Use only the ActionMailer instrumentation 
+  c.use 'OpenTelemetry::Instrumentation::ActionMailer'
+  # Use the ActionMailer instrumentation along with the rest of the Rails-related instrumentation
+  c.use 'OpenTelemetry::Instrumentation::Rails'
+end
+```
+
+Alternatively, you can also call `use_all` to install all the available instrumentation.
+
+```ruby
+OpenTelemetry::SDK.configure do |c|
+  c.use_all
+end
+```
+
+## Active Support Instrumentation
+
+This instrumentation relies entirely on `ActiveSupport::Notifications` and registers a custom Subscriber that listens to relevant events to report as spans.
+
+See the table below for details of what [Rails Framework Hook Events](https://guides.rubyonrails.org/active_support_instrumentation.html#action-mailer) are recorded by this instrumentation:
+
+| Event Name | Creates Span? | Notes |
+| - | - | - |
+| `deliver.action_mailer` | :white_check_mark: | Creates an span with kind `internal` and email content and status|
+| `process.action_mailer` | :x: | Lack of useful info so ignored |
```

Comment:
Not sure whether to open a new issue, but wanted to try here first.

We had added instrumentation for `ActionMailer` ourselves using `OpenTelemetry::Instrumentation::ActiveSupport.subscribe`, but had added `process.action_mailer` as well. The span itself doesn't have interesting attributes but it helps wrap everything necessary to render the email in a common span. We find this very useful. For example:

![image](https://github.com/user-attachments/assets/5cc05518-b4f7-4aa6-8a22-59d5dcd29e2c)

Helps see how long it takes to render the email vs actually deliver it:

![image](https://github.com/user-attachments/assets/294aa2c5-3d11-4d1f-a51d-cdc71c59ec00)

Would you be open to adding this as well?

Reply:
We generally add functionality based on our user need and in many cases we tend to add features that we (maintainers) regularly use in production. 

We are happy to review any contributions you'd like to submit to the community and look forward to your help.